pandas/core/internals.py

```diff
@@ -1821,6 +1824,130 @@ def _unstack(self, unstacker_func, new_columns):
         return blocks, mask
 
 
+class ExtensionBlock(NonConsolidatableMixIn, Block):
```

Comment:
this should NOT be in this file, make a dedicated namespace. pandas.core.internals.block or something

Reply:
Why would it not be in this file? All the other blocks are. Note that `ExtensionBlock` is not ever going to be public, only `ExtensionArray` and `ExtensionDtype`.

Reply:
this is exactly the point, this diff is already way too big. let's do a pre-cursor PR to split Block and manger (IOW internals into 2 files).

Reply:
By splitting the file first, the diff here will not be smaller